packages/notification-center/src/hooks/useFetchOrganization.ts

```diff
@@ -1,17 +1,32 @@
-import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query';
 import { IOrganizationEntity } from '@novu/shared';
 
 import { useNovuContext } from './useNovuContext';
-import { ORGANIZATION_QUERY_KEY } from './queryKeys';
+import { FEED_UNSEEN_COUNT_QUERY_KEY, ORGANIZATION_QUERY_KEY, UNSEEN_COUNT_QUERY_KEY } from './queryKeys';
+import { useEffect, useState } from 'react';
 
 export const useFetchOrganization = (
   options: UseQueryOptions<IOrganizationEntity, Error, IOrganizationEntity> = {}
 ) => {
-  const { apiService, isSessionInitialized, fetchingStrategy } = useNovuContext();
+  const { apiService, isSessionInitialized, fetchingStrategy, subscriberId, applicationIdentifier } = useNovuContext();
+  const [intialize, setInitialize] = useState(false);
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    if (intialize) {
+      queryClient.refetchQueries(UNSEEN_COUNT_QUERY_KEY, { exact: false });
+      queryClient.refetchQueries(FEED_UNSEEN_COUNT_QUERY_KEY, { exact: false });
+      queryClient.refetchQueries(ORGANIZATION_QUERY_KEY, { exact: false });
+    }
+  }, [subscriberId, applicationIdentifier]);
```

Comment:
This hook is supposed to be only responsible for fetching the organization. Now it has additional logic that refetches other queries after we fetched org. I think this is not a good place for this logic to be and I wouldn't expect that if I will debug it. And why do we only refetch these and not for ex. the user preferences?

---

I haven't spend much time digging into the code, but I feel like we should do this in another way. 
In simple words, we just need to update the "query keys" with the `subscriberId, subscriberHash, applicationIdentifier` like the `useSession` does, so any changes to these fields will create the new "query keys" resulting in new fetches for all queries (if the cache is stale).

Please ensure that all the query hooks have these fields in the keys and that the WS session is also re-initialized.